clients/src/main/java/org/apache/kafka/common/requests/JoinGroupResponse.java

```diff
@@ -65,7 +66,17 @@ public JoinGroupResponse(short errorCode,
                              String memberId,
                              String leaderId,
                              Map<String, ByteBuffer> groupMembers) {
-        super(new Struct(CURRENT_SCHEMA));
+        this(CURRENT_VERSION, errorCode, generationId, groupProtocol, memberId, leaderId, groupMembers);
```

Comment:
Sigh. Using the current version is actually useful for test cases on the client side. Our patterns for managing the versions of request/response need some work. Rather than depending on choosing the right constructor to get the right version, I'd rather have static factory methods (e.g. `JoinGroupResponse.currentVersion(args)` or `JoinGroupResponse.byVersion(int version, args)`), which seems less error prone. If you're willing to commit it, we can start a new convention right here!